channeldb/kvdb/etcd/stm.go

```diff
@@ -411,6 +421,13 @@ func (s *stm) Get(key string) ([]byte, error) {
 
 	if len(kvs) > 0 {
 		return []byte(kvs[0].val), nil
+	} else {
+		// Add assertion to the read set which will extend our commit
+		// constraint such that the commit will fail if the key is
+		// present in the database.
+		s.rset[key] = stmGet{
+			rev: 0,
```

Comment:
if a bucket is created and then deleted, it reads to me that this check will always fail where I think bbolt would permit it. not sure how we want to handle that, tho i think it's more of an unlikely edge case.

Reply:
It is actually doesn't fail. Added a test to demonstrate.

Reply:
i had a different scenario mind, i've created the test here: https://paste.ubuntu.com/p/Jwr85nCnH5/

i confirmed locally that this passes, as well as the ordering where a value is created, deleted, and then bucket with the same name is created after